src/WormholeJoin.sol

```diff
@@ -154,14 +154,14 @@ contract WormholeJoin {
             return;
         }
 
-        uint256 fee = vatLive ? FeesLike(fees[wormholeGUID.sourceDomain]).getFees(wormholeGUID, line_, debt_) : 0;
-        require(fee <= maxFee, "WormholeJoin/max-fee-exceed");
-
         uint256 amtToTake = _min(
                                 pending,
                                 uint256(int256(line_) - debt_)
                             );
 
+        uint256 fee = vatLive ? FeesLike(fees[wormholeGUID.sourceDomain]).getFees(wormholeGUID, line_, debt_, amtToTake) : 0;
+        require(fee <= maxFee * amtToTake / pending, "WormholeJoin/max-fee-exceed");
```

Comment:
What about passing `maxFeePercentage` (in BPS) as argument to the function instead of `maxFee` and do `require(fee <= maxFeePercentage * amtToTake / 10000)`. Functionally equivalent but I feel that it would make it a bit more explicit as to how the max fee requirement operates.

Reply:
Ah yes I like this better.